crates/metadata/src/specs.rs

```diff
@@ -942,6 +1024,21 @@ impl<F> EventParamSpec<F>
 where
     F: Form,
 {
+    /// Creates a new event paramater specification builder from a custom type.
+    pub fn new_custom(label: F::String, ty: TypeSpec<F>) -> EventParamSpecBuilder<F> {
```

Comment:
`EventParamSpecBuilder` already has `of_type` to mutate the type, so you don't need this.

Reply:
The reason for it was that `Type::new()` is currently only implemented for `MetaForm`, but this should be doable

Reply:
Yes we should open up any idiomatic `new` constructors to be generic